File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/data_structures/testMap.m
% testMap.m

function testMap
	aMap = DpMap;
	displayMap(aMap);

	aMap.put(1, -1);
	displayMap(aMap);

	aMap.put(2, -2);
	displayMap(aMap);
end

function displayMap(aMap)
	fprintf('displayMap(aMap): ');
	if aMap.contains(1)
		fprintf('[1, %d] ', aMap.get(1));
	end
	if aMap.contains(2)
		fprintf('[2, %d] ', aMap.get(2));
	end
	fprintf('\n');
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/data_structures/test_map.cpp
// test_map.cpp

#include <iostream>
#include "dp_map.h"

template <typename Key, typename Value>
inline
void display_map(const dp_map<Key, Value> & a_map)
{
	std::cout << "display_map(a_map): ";
	if (a_map.contains(1))
	{
		std::cout << "[1, " << a_map.get(1) << "] ";
	}
	if (a_map.contains(2))
	{
		std::cout << "[2, " << a_map.get(2) << "] ";
	}
	std::cout << std::endl;
}

int main()
{
	dp_map<int, int> a_map;
	display_map(a_map);

	a_map.put(1, -1);
	display_map(a_map);

	a_map.put(2, -2);
	display_map(a_map);

	return 0;
}
